usl_pipeline/cloud_functions/main.py

```diff
@@ -329,6 +329,50 @@ def delete_flood_scenario_metadata(cloud_event: functions_framework.CloudEvent)
         metastore.FloodScenarioConfig.delete(db, cloud_event.data["name"])
 
 
+@functions_framework.cloud_event
+@_retry_and_report_errors()
+def build_and_upload_simulation_chunk(
+    cloud_event: functions_framework.CloudEvent,
+) -> None:
+    """Creates a simulation chunk and uploads it to GCS.
+
+    This function is triggered when files containing raw output data for a simulation
+    is uploaded to GCS. It will load the file into simluation chunks bucket and
+    write a new simluation entry to metastore.
+    """
+    file_name = pathlib.PurePosixPath(cloud_event.data["name"])
+    bucket_name = cloud_event.data["bucket"]
+
+    # For AtmoML, only 3rd domain (500m) output files will be used
+    if re.search(file_names.WRF_DOMAIN3_NC_REGEX, file_name.name):
+        storage_client = storage.Client()
+        db = firestore.Client()
+
+        bucket = storage_client.bucket(bucket_name)
+        file_blob = bucket.blob(str(file_name))
+        blob_path = pathlib.PurePosixPath(cloud_event.data["name"])
+
+        # For WRF, we can treat each snapshot output file as its own chunk. If
+        # multiple snapshots must be processed together, we can choose to group
+        # the files together in a TAR before uploading.
+        with file_blob.open("rb") as fd:
+            simulation_chunk_bucket = storage_client.bucket(
+                cloud_storage.SIMULATION_CHUNKS_BUCKET
+            )
+            simulation_chunk_bucket.blob(str(file_name)).upload_from_file(fd)
```

Comment:
It looks like this is a straight copy of the file?
[copy_blob](https://gcloud.readthedocs.io/en/latest/storage-buckets.html#google.cloud.storage.bucket.Bucket.copy_blob)
should be able to do this more cheaply.

I'd also ask if we could upload directly to the simulations bucket rather than leave two copies of the same file in different buckets.

Reply:
Ack  on the two copies - I have it set up this way to mirror what we're doing for the [features](https://github.com/UrbanSystemsLab/climateiq-cnn/blob/main/usl_pipeline/cloud_functions/main.py#L151) to make it more predictable (https://miro.com/app/board/uXjVNgvVs2Q=/)

The other option if we want to just keep one copy of the file is to just write the Simulation entry when we [build the wrf label matrix](https://github.com/UrbanSystemsLab/climateiq-cnn/blob/main/usl_pipeline/cloud_functions/main.py#L335). WDYT

Reply:
creating the entry when we create the label matrix sounds good to me